src/main/java/smile/identity/core/Signature.java

```diff
@@ -1,181 +1,31 @@
-        Long timestamp = System.currentTimeMillis();
-        return generate_sec_key(timestamp);
-    }
-    
-    public String getSecKey(Long timestamp) throws Exception {
-        JSONObject json = (JSONObject) new JSONParser().parse(generate_sec_key(timestamp));
-        return (String) json.get(SEC_KEY);
-    }
-
-    public Boolean confirm_sec_key(String timestamp, String secKey) throws Exception {
-        String toHash = Integer.parseInt(partnerId) + ":" + timestamp;
-        
-        try {
-            MessageDigest md = MessageDigest.getInstance(SHA_256);
-            md.update(toHash.getBytes());
-            byte[] hashed = md.digest();
-            String hashSignature = bytesToHexStr(hashed);
-
-            String[] arrOfStr = secKey.split("\\|");
-            String encrypted = arrOfStr[0];
-
-            PublicKey publicKey = loadPublicKey(apiKey);
-
-            byte[] decodedBytes = Base64.getDecoder().decode(encrypted);
-            String decrypted = decryptString(publicKey, decodedBytes);
-
-            return decrypted.equals(hashSignature);
-        } catch (Exception e) {
-            throw e;
-        }
-    }
-
-    public static Boolean useSignature(JSONObject options) {
-        if (options.containsKey(Signature.SIGNATURE_KEY)) {
-            return (Boolean) options.get(Signature.SIGNATURE_KEY);
-        }
-        return true;
-    }
 
-    private static PublicKey loadPublicKey(String apiKey) throws GeneralSecurityException, IOException {
-        apiKey = apiKey.replace("-----BEGIN PUBLIC KEY-----", "").replace("-----END PUBLIC KEY-----", "").replace("\n", "").replace("\r", "").trim();
-        apiKey = apiKey.replaceAll(" ", "");
-        byte[] data = Base64.getDecoder().decode((apiKey.getBytes()));
-        X509EncodedKeySpec spec = new X509EncodedKeySpec(data);
-        return KeyFactory.getInstance("RSA").generatePublic(spec);
+    public SignatureKey generateSignature() {
+        return generateSignature(System.currentTimeMillis());
     }
 
-    private static byte[] encryptString(PublicKey key, String plaintext) throws NoSuchAlgorithmException, NoSuchPaddingException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
-        Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");
-        cipher.init(Cipher.ENCRYPT_MODE, key);
-        return cipher.doFinal(plaintext.getBytes());
+    public SignatureKey generateSignature(Long timestamp){
+        return new SignatureKey(timestamp, this.partnerId, this.apiKey);
     }
 
-    private static String decryptString(PublicKey key, byte[] encrypted) throws NoSuchAlgorithmException, NoSuchPaddingException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
-        Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");
-        cipher.init(Cipher.DECRYPT_MODE, key);
-        return new String(cipher.doFinal(encrypted));
+    public Boolean validateSignature(Long timestamp, String signature){
```

Comment:
Is this method needed when `confirmSignature` has the same fingerprint? 🤔

Reply:
lol yeah, just removed it 👍🏾